pkg/migration/controllers/migration.go

```diff
@@ -535,8 +543,44 @@ func (m *MigrationController) handle(ctx context.Context, migration *stork_api.M
 	return nil
 }
 
+func (m *MigrationController) getMigrationNamespaces(ctx context.Context, migration *stork_api.Migration) ([]string, error) {
+	var migrationNamespaces []string
+	uniqueNamespaces := make(map[string]bool)
+
+	for _, ns := range migration.Spec.Namespaces {
+		if _, ok := uniqueNamespaces[ns]; !ok {
+			migrationNamespaces = append(migrationNamespaces, ns)
+		}
+		uniqueNamespaces[ns] = true
+	}
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	for key, val := range migration.Spec.NamespaceSelectors {
+		label := key + "=" + val
+		namespaces, err := clientset.CoreV1().Namespaces().List(ctx, metav1.ListOptions{LabelSelector: label})
+		if err != nil {
+			return nil, err
+		}
+		for _, namespace := range namespaces.Items {
+			if _, ok := uniqueNamespaces[namespace.GetName()]; !ok {
+				migrationNamespaces = append(migrationNamespaces, namespace.GetName())
+			}
+			uniqueNamespaces[namespace.GetName()] = true
```

Comment:
Another approach could have been to only populate `uniqueNamespaces` map with namespaces and at the end create the list of keys and return that. The current approach is fine too.

Reply:
Updated as per suggestion.